attiny/attiny.ino

```diff
@@ -35,22 +40,72 @@ enum State {
   PI_WDT_FAILED             // WDT for RPi failed. Turn off and on.
 };
 
+#define I2C_READ_REG_LEN 1
+#define I2C_READ_BATTERY_VOLTAGE 0x20
+
+volatile byte i2cReadRegVal = 0x00;
+volatile byte i2cReadRegs[I2C_READ_REG_LEN] =
+{
+    I2C_READ_BATTERY_VOLTAGE,
+};
+
 State state = PI_POWERED;
 
 void setup() {
+  pinMode(BATTERY_VOLTAGE_PIN, INPUT);
   pinMode(PI_POWER_PIN, OUTPUT);
   pinMode(POWER_LED, OUTPUT);
+  digitalWrite(PI_POWER_PIN, LOW);
+  digitalWrite(POWER_LED, LOW);
+  checkBattery();
   digitalWrite(POWER_LED, LOW);
   initTimer1();
   sei(); // enable interupts
   TinyWireS.begin(I2C_SLAVE_ADDRESS);
   TinyWireS.onReceive(receiveEvent);
   TinyWireS.onRequest(requestEvent);
   setBlinks(2);
-  wdt_enable(WDTO_8S); // enable 8 second WDT
 }
 
-void loop() { 
+void checkBattery() {
+  int a = analogRead(BATTERY_VOLTAGE_PIN);
+  if (425 <= a && a <= 645) {           // Check if battery is between 5.5V and 6.9V
```

Comment:
constants for these values?

Reply:
please add a comment that below 5.5V could be that the board is being powered by 5V supply